packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -468,19 +468,68 @@ export class SDKClient {
     const transactionType = transaction.constructor.name;
     const requestIdPrefix = formatRequestIdMessage(requestId);
     const currentDateNow = Date.now();
+    let gasUsed: any = 0;
+    let transactionFee: number = 0;
+    let transactionId: string = '';
     try {
-      const shouldLimit = this.hbarLimiter.shouldLimit(currentDateNow, SDKClient.transactionMode, callerName);
-      if (shouldLimit) {
+      if (this.hbarLimiter.shouldLimit(currentDateNow, SDKClient.transactionMode, callerName))
         throw predefined.HBAR_RATE_LIMIT_EXCEEDED;
-      }
 
       this.logger.info(`${requestIdPrefix} Execute ${transactionType} transaction`);
-      const resp = await transaction.execute(this.clientMain);
+      const transactionResponse = await transaction.execute(this.clientMain);
```

Comment:
Q: Are you just extracting `executeGetTransactionRecord()`?
Why not just call it here as all this logic is already covered?
No longer calling `executeGetTransactionRecord()` was a regression.
```suggestion
      const transactionResponse = await transaction.execute(this.clientMain);
      executeGetTransactionRecord(transactionResponse, transactionType, callerName, interactingEntity, requestId);
```

Reply:
Updated with new changes. `executeGetTransactionRecord()` is now tweaked and reused.